fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/source/deserializer/RowDataDeserializationSchema.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * <p>Usage example:
+ *
+ * <pre>{@code
+ * RowType rowType = ...; // Define your row type schema
+ * RowDataDeserializationSchema schema = new RowDataDeserializationSchema(rowType);
+ * FlussSource<RowData> source = FlussSource.builder()
+ *     .setDeserializationSchema(schema)
+ *     .build();
+ * }</pre>
+ *
+ * @since 0.7
+ */
+@PublicEvolving
+public class RowDataDeserializationSchema implements FlussDeserializationSchema<RowData> {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Converter responsible for transforming Fluss row data into Flink's {@link RowData} format.
+     * Initialized either in constructor or during {@link #open(InitializationContext)}.
+     */
+    private transient FlussRowToFlinkRowConverter converter;
+
+    /**
+     * Optional row type provided via constructor. If null, row type will be inferred from context.
+     */
+    private final RowType rowType;
+
+    /**
+     * Creates a new {@link RowDataDeserializationSchema} that will infer the row type from the
+     * context during initialization.
+     */
+    public RowDataDeserializationSchema() {
+        this.rowType = null;
+    }
+
+    /**
+     * Creates a new {@link RowDataDeserializationSchema} with the specified row type.
+     *
+     * @param rowType The Fluss row type that describes the structure of the input data
+     */
+    public RowDataDeserializationSchema(RowType rowType) {
+        this.rowType = rowType;
+        this.converter = new FlussRowToFlinkRowConverter(rowType);
+    }
+
+    /**
+     * Initializes the deserialization schema.
+     *
+     * <p>This implementation doesn't require any initialization.
+     *
+     * @param context Contextual information for initialization
+     * @throws Exception if initialization fails
+     */
+    @Override
+    public void open(InitializationContext context) throws Exception {
+        if (converter == null) {
+            RowType schemaToUse = rowType != null ? rowType : context.getRowSchema();
+            this.converter = new FlussRowToFlinkRowConverter(schemaToUse);
+        }
+    }
+
+    /**
+     * Deserializes a {@link LogRecord} into a Flink {@link RowData} object.
+     *
+     * @param record The Fluss LogRecord to deserialize
+     * @return The deserialized RowData
+     * @throws Exception If deserialization fails or if the record is not a valid {@link ScanRecord}
+     */
+    @Override
+    public RowData deserialize(LogRecord record) throws Exception {
+        if (converter == null) {
+            throw new IllegalStateException(
+                    "Converter not initialized. The open() method must be called before deserializing records.");
+        }
+        return converter.toFlinkRowData(record);
+    }
+
+    /**
+     * Returns the TypeInformation for the produced {@link RowData} type.
+     *
+     * @return TypeInformation for RowData class
+     */
+    @Override
+    public TypeInformation<RowData> getProducedType() {
```

Comment:
I believe we can introduce a method called `getProducedType(RowType rowSchema)` instead of extending `ResultTypeQueryable` to allow the `RowType` to be passed in before the `open()` method is invoked. This approach works well because this method would only be called internally by the Fluss source itself, not by Flink. Additionally, this eliminates the need to introduce a constructor that accepts `RowType` (if I understand correctly, this is the purpose of the constructor you proposed).